ludwig/data/preprocessing.py

```diff
@@ -1191,7 +1193,11 @@ def cast_columns(dataset_df, features, global_preprocessing_parameters,
 
 
 def build_metadata(
-        metadata, dataset_cols, features, global_preprocessing_parameters, backend
+        metadata: Dict[str, Any],
+        dataset_cols: Dict[str, dd.Series],
```

Comment:
This can also be a Pandas Series. One thing we may want to do is create our own typing module for Ludwig that has these generic DataFrame abstractions. One thing we need to incorporate here is the possibility that Dask is not installed (since it's optional), so that should be included in the type.

Reply:
Done!